base-networking-azure-go/main.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	network "github.com/pulumi/pulumi-azure-native/sdk/go/azure/network"
+	resources "github.com/pulumi/pulumi-azure-native/sdk/go/azure/resources"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+func main() {
+	pulumi.Run(func(ctx *pulumi.Context) error {
+		// Create a new resource group
+		resourceGroup, err := resources.NewResourceGroup(ctx, "resourceGroup", nil)
+		if err != nil {
+			return err
+		}
+
+		// Create a new virtual network
+		virtualNetwork, err := network.NewVirtualNetwork(ctx, "virtualNetwork", &network.VirtualNetworkArgs{
+			AddressSpace: &network.AddressSpaceArgs{
+				AddressPrefixes: pulumi.StringArray{
+					pulumi.String("10.0.0.0/16"),
+				},
+			},
+			ResourceGroupName: resourceGroup.Name,
+		})
+		if err != nil {
+			return err
+		}
+
+		// Create three subnets in the virtual network
```

Comment:
Any way to use a for loop for this?

Reply:
Yeah, I can rewrite the Go/Python/TypeScript templates to use a loop. Good call.

Reply:
Addressed via bd2863c; the Go, Python, and TypeScript templates now use loops to create the subnets.